### ISSUE TITLE
output sometimes flash and disappear

### ISSUE THREAD

**Issue by longkai** [nushell/nushell#13570]
### Describe the bug

occasionally, the nushell output with abnormal chars like `^[[15;12R` or disappear.

https://github.com/user-attachments/assets/b3aa12e9-d795-4610-9fd7-d984fb984f63

### How to reproduce

not sure how to reproduce but most likely when you use the terminal opened yesterday it will result it.

### Expected behavior

normal output

### Configuration

❯ version | transpose key value | to md --pretty
| key                | value                                          |
| ------------------ | ---------------------------------------------- |
| version            | 0.96.1                                         |
| major              | 0                                              |
| minor              | 96                                             |
| patch              | 1                                              |
| branch             |                                                |
| commit_hash        |                                                |
| build_os           | macos-aarch64                                  |
| build_target       | aarch64-apple-darwin                           |
| rust_version       | rustc 1.79.0 (129f3b996 2024-06-10) (Homebrew) |
| cargo_version      | cargo 1.79.0                                   |
| build_time         | 2024-07-29 23:31:36 +00:00                     |
| build_rust_channel | release                                        |
| allocator          | mimalloc                                       |
| features           | default, sqlite, system-clipboard, trash       |
| installed_plugins  |                                                |

**Comment by fdncred** [comment]
I've run nushell on macos for years and never seen anything like this. I wonder if this has something to do with your setup like with tmux, zellij, alacritty or other termnial emulators/applications.

The `^[[15;12R` is an ansi escape sequence response to the terminal asking the question about cursor location. The terminal should receive this response and you shouldn't see it.

**Comment by longkai** [comment]
I use iterm2 on local macOS without tmux or any related managed tools.

Besides, I also use vscode remote terminal for daily dev. Both have same `env` and `config`.

The configs are simple with some env and aliases with additional `starship` and `carapace` few lines config.

Any suggestions?

**Comment by fdncred** [comment]
I'd start with use the default configs and then at the end source files that override the settings you want.

Then I'd try a few other terminals to see if you get the same output. wezterm, ghostty if you're in the beta, some people like warp.

Then I'd try without starship or carapace.

The point is trying to narrow down where this mysterious behavior is coming from by trying other things.

**Comment by MMukundi** [comment]
Hi all!
I've been running into this issue as well, so I tried the same in macos terminal, warp, and alacritty with no nushell config, and the issue seems to persist; namely, the shell can function normally after several commands, but after `ansi cursor_position` is printed, all output is truncated oddly. 
![Issue in barebones nushell](https://github.com/user-attachments/assets/2555ac48-d4d8-4990-a449-e8b93d16d06e)

**Comment by fdncred** [comment]
Good find. I'm guessing that `ansi cursor_position` somehow puts crossterm in raw mode and then all bets are off.

**Comment by fusion44** [comment]
I can trigger this reliably with the following configuration:
Host: `NixOS; Alacritty; Nushell; Zellij`
I the SSH into a VM running the same version of Nushell from within the host Zellij and the above problem shows up.
The problem won't show up in the following conditions:
* Host with Zellij and VM in Bash mode
* Host without Zellij and VM in Nushell or Bash

**Comment by suimong** [comment]
> I can trigger this reliably with the following configuration: Host: `NixOS; Alacritty; Nushell; Zellij` I the SSH into a VM running the same version of Nushell from within the host Zellij and the above problem shows up. The problem won't show up in the following conditions:
> 
>     * Host with Zellij and VM in Bash mode
> 
>     * Host without Zellij and VM in Nushell or Bash
> 
> 
> _edit_ - This problem shows up in 0.97 first. When I revert back to 0.96.1, everything is fine.

Same here. I tried some more terminal emulators (Alacritty, GNOME Terminal, Kitty) with zellij started manually from nushell, then `ls` just flashes and disappears. 

Also, while haven't done a precise bisect, but I noticed this after upgrading nushell to 0.98.0

**Comment by fdncred** [comment]
A good bisect would be helpful.

**Comment by suimong** [comment]
I investigated a bit more, and it seems that the flashing behavior can be repeatedly reproduced by setting `$env.use_kitty_protocol = true` in a nushell session inside zellij:

![image](https://github.com/user-attachments/assets/e29f571c-5dee-46f0-9ccd-37d928569137)

This is true for the following scenarios:
- nushell version 0.96.1 and 0.98.0 (I only tested these two because the investigation started as a git bisect, and soon I realized that it might not be entirely due to the version difference, so I stopped testing builds in between)
- whether `$env.use_kitty_protocol` is initially (i.e. in `config.nu`) set to true or false is irrelevant. The behavior can be always reproduced by setting this config entry later.
- terminal emulators tested: Alacritty (whose support for kitty protocol seems undocumented), Kitty, VSCode terminal (xterm.js, does not support kitty protocol).

**Comment by suimong** [comment]
I have no clue how to debug further. @fdncred could you advise?

**Comment by fdncred** [comment]
nope, i'm not familiar with the kitty protocol. I'm not sure how to debug it but I do know that it would probably be in reedline.

**Comment by suimong** [comment]
> I investigated a bit more, and it seems that the flashing behavior can be repeatedly reproduced by setting $env.use_kitty_protocol = true in a nushell session inside zellij:


On nushell v0.101.0 (haven't upgraded to v0.102.0), I can no longer reproduce this issue. Setting `$env.config.use_kitty_protocol = true` within zellij, nushell behaves normally now. 

Maybe others who encountered this in different ways can test again?

**Comment by PitiBouchon** [comment]
So I had the issue that when there was a lot of characters printed in the terminal (using a verbose command) I would sometimes have some ansi escape sequence printed after nearly every command. Sometimes this ansi escape sequence can clear the screen which seems like a flash.

![Image](https://github.com/user-attachments/assets/ca0f253d-e708-4479-bf6d-b48d7b388511)

 Using a simple script :
```nu
rm -f random.txt
let x = (random binary 1000 | decode utf-8)
print $x
$x | save random.txt
```

and having in my config :
```nu
$env.config = {
    use_kitty_protocol: true
}
```

I can reproduce this within a few try on alacritty with `nu --version = 0.103.0`  and `alacritty --version = 0.15.1` (so recent versions).

My `random.txt` file example :
[random.txt](https://github.com/user-attachments/files/19875004/random.txt)

**Comment by PitiBouchon** [comment]
It seems even with kitty protocol disabled I sometimes (when CTRL+C a `cargo clippy` with many warnigs) get some ansi characters at the end of every commands (the same way as described in the comment above), but I cannot reproduce it using this method

**Comment by fusion44** [comment]
Is anyone who experiences this problem using [Atuin](https://atuin.sh/)? I experienced this problem a lot during normal development today. I've then disabled Atuin and the problem seems to be gone for now.

**Comment by PitiBouchon** [comment]
After some investigation you only need to do a `char --integer (27) (90)` to trigger the issue, and it comes (every new prompt) from `kitty_protocol_available` (in reedline)

Cannot reproduce this on kitty, foot and the gnome terminal now, and as mentioned in the code comments, only kitty / foot and wezterm are supported and not things like alacritty, rio or ghostty which I use.

To improve this, `kitty_protocol_available`  could be called once on struct initialization instead 🤷 to avoid side-effect during the usage of the shell
